### ISSUE TITLE
`panic: interface conversion: interface {} is map[string]interface {}, not string` in version 2.90.0

### ISSUE THREAD

**Issue by maxbog** [hashicorp/terraform-provider-azurerm#14707]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v1.1.2
on windows_amd64
```

Provider version: 2.90.0

### Affected Resource(s)

* `azurerm_api_management_api_operation`

### Terraform Configuration Files

```hcl
resource "azurerm_api_management_api_operation" "list_sites" {
  operation_id = "GetSites"
  api_name = var.api_instance.name
  api_management_name = var.api_management_name
  resource_group_name = var.resource_group.name
  display_name = "List sites"
  method = "POST"
  url_template = "/sites/graph"
  description = "List Sites"
}
```

### Panic Output
```
│    1: resource "azurerm_api_management_api_operation" "list_sites" {
│
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-azurerm_v2.90.0_x5.exe plugin:

panic: interface conversion: interface {} is map[string]interface {}, not string

goroutine 1348 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz.FlattenApiManagementOperationParameterExampleContract(0xc001bf5890)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/apimanagement/schemaz/api_management.go:423 +0x3fc        
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement/schemaz.FlattenApiManagementOperationRepresentation(0xc00139ae70)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/apimanagement/schemaz/api_management.go:196 +0x1db        
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement.flattenApiManagementOperationResponseContract(0xc00139ae40)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/apimanagement/api_management_api_operation_resource.go:359
 +0x245
github.com/hashicorp/terraform-provider-azurerm/internal/services/apimanagement.resourceApiManagementApiOperationRead(0xc001d2a100, {0x52646c0, 0xc00049b180})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/apimanagement/api_management_api_operation_resource.go:213
 +0x6b6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x5b3071a, {0x6235c78, 0xc001d14880}, 0x24, {0x52646c0, 0xc00049b180})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.g
o:346 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000434b60, {0x6235c78, 0xc001d14880}, 0xc0011b92b0, {0x52646c0, 0xc00049b180})
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.g
o:635 +0x35b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0013e6330, {0x6235c78, 0xc001d14880}, 0xc001d14900)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provi
der.go:576 +0x534
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc00028e780, {0x6235d20, 0xc001d1aa50}, 0xc0014e3b60)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.g
google.golang.org/grpc.(*Server).handleStream(0xc000230fc0, {0x6284388, 0xc0003bc600}, 0xc001d1c7e0, 0x0)
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294

Error: The terraform-provider-azurerm_v2.90.0_x5.exe plugin crashed!
```

### Expected Behaviour

Plan should have worked. It still works in 2.89.0

### Actual Behaviour

Plan causes panic

### Steps to Reproduce

1. Upgrade azurerm provider to 2.90.0
2. `terraform apply`

### References

Looks like this change may have caused the problem:
* #14312

**Comment by xuzhang3** [comment]
The APIM `example.value` is an generic value, this value can be anything, a primitive value, or an object(https://github.com/Azure/azure-sdk-for-go/blob/main/services/apimanagement/mgmt/2021-08-01/apimanagement/models.go#L10236). Simply converting the response value to string is not enough.

**Comment by maxbog** [comment]
#14707 is a duplicate

**Comment by maxbog** [comment]
@mybayern1974 Any chance of someone reviewing the fix (#14848)?. This breaks API Management operation.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #14848. As such, I am going to mark this issue as closed.
